src/Workspaces/Core/Portable/Diagnostics/Extensions.cs

```diff
@@ -21,6 +22,8 @@ namespace Microsoft.CodeAnalysis.Diagnostics;
 
 internal static partial class Extensions
 {
+    private static readonly ConditionalWeakTable<Project, AsyncLazy<Checksum>> s_projectToDiagnosticChecksum = new();
```

Comment:
>ConditionalWeakTable [](http://example.com/codeflow?start=28&length=20)

For my understanding:

I thought generally it was best to avoid CWTs unless it was quite difficult not to do so as the add extra strain to GC. Is this not the case?